src/profile-logic/tracks.js

```diff
@@ -610,6 +610,13 @@ function _isThreadIdle(profile: Profile, thread: Thread): boolean {
     return false;
   }
 
+  if (thread.samples.length === 0) {
+    // This is a profile without any sample(taken with no periodic sapling mode)
+    // and we can't take a look at the samples to decide whether that thread is
+    // active or not. So we are checking if we have a paint marker instead.
+    return _isThreadWithNoPaint(thread);
```

Comment:
Not sure about the other threads but at least having the processes that are non-idle is useful for users. Without this we are just hiding everything. But yeah, we can find a way for other threads as well. responsiveness is very dependent on the machine though, it can be hard to create a heuristic that applies to all machines.

Reply:
Created #2386 for that.